loop.go

```diff
@@ -62,3 +70,16 @@ Wait:
 		}
 	}
 }
+
+// Shutdown is used to cleanly shutdown go-statsite, flushing all metrics before
+// exiting.
+func Shutdown() {
+	if !enabled {
+		return
+	}
+	// No more metrics, disable the flusher and return
+	enabled = false
+	close(statQueue)
```

Comment:
it gets tricky closing a channel here, you should only close a channel from the sender.  If for some reason there is a go routine that hasn't been execute that runs after this close there will be a panic.

What I've seen in blogs and such is a DONE channel that is used to coordinate things